app/src/main/java/com/android/sample/ui/authentication/SignIn.kt

```diff
@@ -141,11 +131,44 @@ fun SignInScreen(
   } else {
     // When user is signed in, check if they have an account
     LaunchedEffect(userAccount) {
-      if (userAccount != null) {
+      showLoadingDialog = true // Show loading screen
+      Firebase.auth.currentUser?.uid?.let { userId ->
+        checkUserAccountAndNavigate(userId, userAccountViewModel, navigationActions, scope) {
+          showLoadingDialog = false // Hide loading screen
+        }
+      }
+    }
+  }
+
+  // Show Loading Dialog if `showLoadingDialog` is true
+  if (showLoadingDialog) {
+    LoadingDialog(onDismissRequest = { showLoadingDialog = false })
+  }
+}
+
+fun checkUserAccountAndNavigate(
+    userId: String,
+    userAccountViewModel: UserAccountViewModel,
+    navigationActions: NavigationActions,
+    scope: CoroutineScope,
+    onComplete: () -> Unit
+) {
+  userAccountViewModel.getUserAccount(userId)
+
+  scope.launch {
+    delay(600) // Wait for data to be fetched
```

Comment:
Is there a way to wait for response and not a hardcoded value ? Otherwise it is ok like that.